aries_cloudagent/protocols/present_proof/dif/pres_exch_handler.py

```diff
@@ -416,59 +419,60 @@ async def filter_constraints(
                     credential_dict=credential_dict, constraints=constraints
                 )
                 derive_key = self.get_derive_key_credential_subject_id([credential])
-                if not derive_key:
-                    raise DIFPresExchError(
-                        "Applicable credential to derive"
-                        " does not contain credentialSubject.id"
+                async with self.profile.session() as session:
+                    wallet = session.inject(BaseWallet)
+                    derive_suite = await self._get_derive_suite(
+                        wallet=wallet,
+                        issuer_id=derive_key,
                     )
-                else:
-                    async with self.profile.session() as session:
-                        wallet = session.inject(BaseWallet)
-                        derive_suite = await self._get_derive_suite(
-                            wallet=wallet,
-                            issuer_id=derive_key,
-                        )
-                        signed_new_credential_dict = await derive_credential(
-                            credential=credential_dict,
-                            reveal_document=new_credential_dict,
-                            suite=derive_suite,
-                            document_loader=document_loader,
-                        )
-                        credential = self.create_vcrecord(signed_new_credential_dict)
+                    signed_new_credential_dict = await derive_credential(
+                        credential=credential_dict,
+                        reveal_document=new_credential_dict,
+                        suite=derive_suite,
+                        document_loader=document_loader,
+                    )
+                    credential = self.create_vcrecord(signed_new_credential_dict)
             result.append(credential)
         return result
 
     def create_vcrecord(self, cred_dict: dict) -> VCRecord:
         """Return VCRecord from a credential dict."""
-        given_id = cred_dict.get("id")
+        proofs = cred_dict.get("proof") or []
+        proof_types = None
+        if type(proofs) is dict:
+            proofs = [proofs]
+        if proofs:
+            proof_types = [proof.get("type") for proof in proofs]
         contexts = [ctx for ctx in cred_dict.get("@context") if type(ctx) is str]
-
+        if "@graph" in cred_dict:
+            for enclosed_data in cred_dict.get("@graph"):
+                if enclosed_data["id"] == "urn:bnid:_:c14n0":
+                    cred_dict.update(enclosed_data)
+                    del cred_dict["@graph"]
+                    break
+        given_id = cred_dict.get("id")
```

Comment:
Are you sure this will consistently work? I try to stay away from very specific jsonld syntax assumptions as they can be unpredictable IMO.

Reply:
If you're sure this works, maybe add a comment explaining what it does/why it's needed?

Reply:
Based upon the testing I was doing, when deriving a credential, the `@graph` was always something like this:
```
[
		{
			'https://www.w3.org/2018/credentials#credentialSubject': None, 
			'https://www.w3.org/2018/credentials#issuanceDate': None, 
			'https://www.w3.org/2018/credentials#issuer': None, 
			'id': 'urn:bnid:_:c14n0', 
			'type': ['Person', 'PermanentResident']
		}, 
		{
			'credentialSubject': None,
			'id': 'urn:bnid:_:c14n1',
			'issuanceDate': '2020-01-01T12:00:00Z',
			'issuer': 'did:key:zUC765p5CiwfJ3mgVfF5jk3ga4BsudkLggcT1P3cdE6x9tsXwhEBrfsJoHyG7AE5bxoN7C81aoAacwdkiZoCzY49fmSeP1m67wUoPZfE4kBasLNY9RbT9iU2QUXkZFEnMXgWhMb',
			'type': ['PermanentResident', 'VerifiableCredential']
		}
	]
	```
I have to create a VCRecord from this to enclose this credential into the presentation. I have updated the logic to check if the object's `id` startswith ``urn:` and if there is a `credentialSubject`. I think this should consistently work.

Reply:
Ok. However I still think we should solve the underlying problem instead